Source/JavaScriptCore/wasm/js/JSWebAssemblyArray.h

```diff
@@ -84,9 +110,25 @@ class JSWebAssemblyArray final : public JSNonFinalObject {
         }
     }
 
-    void set(VM& vm, uint32_t index, EncodedJSValue value)
-    {
-        switch (m_elementType.kind) {
+    void set(VM& vm, uint32_t index, EncodedJSValue value) {
+        if (m_elementType.type.is<Wasm::PackedType>()) {
+            // `value` is assumed to be an unboxed int64; truncate it to either 8 or 16 bits
+            uint32_t val = (uint32_t) value;
+            switch (*m_elementType.type.as<Wasm::PackedType>()) {
+            case Wasm::PackedType::I8:
+                m_payload8[index] = ((uint8_t) val);
```

Comment:
I think it's sufficient to do one cast here, something like `static_cast<uint8_t>(value)` (`static_cast` preferred because of https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-casts-named).